Sidebar: rename "Plan" section to "Upgrades"

michaeldcain opened the pull request:
Now that site-level purchase management has been added and a "Plan" section created, we have the redundant `Plan > Plans > My Plan` navigation. To house all purchase-related navigation, this PR does the following:

- renames the "Plan" section to "Upgrades"
- moves the plan indicator to the "Plans" navigation item
- adds a (duplicate for now) link to the domain management screen
- renames the "Billing" item to "Purchases"
- within the "Purchases" site-level and account-level screens:
  - updates the formatted header to "Purchases"
  - renames the "Purchases" section nav item to "Active Upgrades"
- Updates the fallback for the unified domains project (see: https://github.com/Automattic/jetpack/pull/18093)

See: pbOQVh-KN-p2

Before | After
------------ | -------------
![before](https://user-images.githubusercontent.com/942359/102799169-23c63800-4380-11eb-8234-ba3909154bbf.png) | ![after](https://user-images.githubusercontent.com/942359/102799182-288aec00-4380-11eb-835b-8239df2522ec.png)


**To test:**
- visit Calypso
- verify that the "Upgrades" section is shown in the sidebar and matches the screenshot above
- verify that the links inside the section correctly navigate to '/plans', '/purchases', and '/domains'
- verify that the sectionNav, formattedHeader, and various pageTitles inside the Purchases screen are accurate
- make sure that the tests and e2es pass

michaeldcain committed:
Update function name that renders domains item

michaeldcain committed:
Update plan section to upgrades section

a8ci18n commented on client/my-sites/sidebar/index.jsx:
```diff
@@ -580,43 +580,40 @@ export class MySitesSidebar extends Component {
+				expanded={ this.props.isUpgradesSectionOpen }
+				title={
+					<>
+						<span className="sidebar__menu-link-text menu-link-text" data-e2e-sidebar="Upgrades">
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **42** times:
`translate( 'Upgrades', { context: 'Help topic'} )` *ES Score: 13*


:information_source: This string already exists with the following **contexts**:
- `Help topic`

 Would it make sense to reuse one of the above?

michaeldcain replied:
@Automattic/i18n: both strings are used as navigation items with a noun context. Should I update the context of the "help topic" string?

yuliyan replied:
@michaeldcain, I think it makes more sense to leave it with "noun" context, even though it will be sent for translation.

fditrapani commented:
Overall, I really like how this looks and think it's a good update. The double domain highlight doesn't look right to me. I think we should decide where we want to show it and show it in one place. I personally think it's fine in upgrades especially since we have a link to it from the general settings screen. That being said, I think this update would benefit from running it as an a/b test to closer monitor the impact on purchases and also possibly a p2 post to open it up for more discussion as well as general awareness.

fditrapani commented:
Any thoughts about updating 'Manage Purchases' in `/me` to `Upgrades` as well?

fditrapani commented:
Another thing I noticed:

![image](https://user-images.githubusercontent.com/6981253/99283289-cda32980-2802-11eb-9177-710ba7528fa2.png)

We should update the `Billing` page title to `Purchases` and `Purchases` tab to `Active upgrades`.

fditrapani commented:
I took a quick look through support to see if there were any references we need to update. There's one here that can be fine as is (because of the link) but would be better if we update it: https://wordpress.com/support/manage-purchases/#upgrading-your-plan.

michaeldcain commented:
> Any thoughts about updating 'Manage Purchases' in /me to Upgrades as well?

@fditrapani, I thought about that, but technically that menu item matches the "Purchases" menu item at the site-level. I think if anything, we'd want to drop the "Manage", which would match the noun naming of the other sidebar items at the account-level.

fditrapani commented:
> I think if anything, we'd want to drop the "Manage", which would match the noun naming of the other sidebar items at the account-level.

Yeah, I think that makes sense. We can drop "Manage" and then update the tabs to match what we have on the site-level.

michaeldcain committed:
Rename account-level purchases sidebar link

michaeldcain committed:
Update account-level section navigation

michaeldcain committed:
Update account-level section title

michaeldcain committed:
Update site-level section nav and title

michaeldcain committed:
Missed one

michaeldcain committed:
Consolidate titles

Commit:
Fix section highlighting

Commit:
Remove deleted file

Commit:
Remove duplicate domains

michaeldcain committed:
Update purchases title to active upgrades

michaeldcain committed:
Use titles object in purchases navigation

michaeldcain committed:
Remove localize from BillingHistory and convert to TS

sirbrillig commented on client/my-sites/purchases/main.tsx:
```diff
@@ -51,26 +52,26 @@ function useLogPurchasesError( message: string ) {
 			);
 		},
-		[ reduxDispatch ]
+		[ reduxDispatch, message ]
```
Nice find. That's an important dependency.

michaeldcain committed:
Update PendingPayments tests to account for nav change

michaeldcain committed:
Move plans indicator to Plans navigation item

michaeldcain committed:
Leave current domains item in place but expand upgrades section

michaeldcain committed:
Update e2es

michaeldcain committed:
Collapse manage section and expand upgrades when Domains is clicked

michaeldcain commented:
@sirbrillig: I've updated the tests, made the changes suggested in the attached P2 post, and updated the PR description. I think this is good to review.

a8ci18n commented on client/my-sites/sidebar/index.jsx:
```diff
@@ -578,50 +586,42 @@ export class MySitesSidebar extends Component {
+			<ExpandableSidebarMenu
+				onClick={ this.toggleSection( SIDEBAR_SECTION_UPGRADES ) }
+				expanded={ this.props.isUpgradesSectionOpen }
+				title={ this.props.translate( 'Upgrades', { context: 'noun' } ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **42** times:
`translate( 'Upgrades', { context: 'Help topic'} )` *ES Score: 13*


:information_source: This string already exists with the following **contexts**:
- `null` (no context)
- `Help topic`

 Would it make sense to reuse one of the above?

michaeldcain replied:
This was discussed pre-rebase and the decision was to leave it with the "noun" context.

sirbrillig commented on test/e2e/lib/components/sidebar-component.js:
```diff
@@ -64,10 +59,15 @@ export default class SidebarComponent extends AsyncBaseContainer {
 
+	async selectDomains() {
+		await this.expandDrawerItem( 'Upgrades' );
+		return await this._scrollToAndClickMenuItem( 'domains' );
```
This is perfectly fine (and it matches the other functions here), but just so _you_ know, there's no need for the `return await functionThatReturnsAPromise()` pattern here since this function is already marked `async` (it will always return a Promise) and therefore just `return functionThatReturnsAPromise()` will have the same effect.

sirbrillig replied:
Documented better here: https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md

sirbrillig reviewed (approved):
Code looks great and testing works as described. This is much more clear.
